mono/metadata/handle.h

```diff
@@ -32,15 +24,33 @@ typedef MonoHandleStorage* MonoHandle;
  * The field obj is not private as there is no way to do that
  * in C, but using a C++ template would simplify that a lot
  */
-struct _MonoHandleStorage {
-	MonoObject *obj;
-};
+typedef struct {
+	MonoObject *__private_obj;
+} MonoHandleStorage;
+
+typedef MonoHandleStorage* MonoHandle;
+
+#include "handle-arena.h"
```

Comment:
You can do
```
  /* here */
  #include "handle-arena.h"
  struct _MonoHandleStorage {
     MonoObject *__private_obj;
  }
  typedef struct _MonoHandleStorage MonoHandleStorage;
  typedef MonoHandleStorage* MonoHandle;

  /* in handle-arena.h */
  typedef struct _MonoHandleStorage MonoHandleStorage;
  typedef MonoHandleStorage* MonoHandle;
```

The stuff in `handle-arena.h` doesn't actually need to see the definition of `struct _MonoHandleStorage` in order to define a typedef to it.  C is funny that way.